File: CSharp/SpecStudioExamplesTestsCSharp/common/PatchTitleTyped.cs
namespace SpecStudioExamplesTestsCSharp.common
{
using System;
using System.Buffers;
using System.Collections.Generic;
using System.Globalization;
using System.Text;
using System.Text.Json;
using production;

    public class PatchTitleTyped
    {
        public string title;

        public PatchTitleTyped(string title)
        {
            this.title = title;
        }

        public PatchTitleString ToPatchTitleString()
        {
            return new PatchTitleString(
                Json.ToText(this.title)
            );
        }

        public static List<PatchTitleString> ToStringList(List<PatchTitleTyped> list)
        {
            var result = new List<PatchTitleString>();
            foreach (var t in list) result.Add(t.ToPatchTitleString());
            return result;
        }

        public static List<PatchTitleTyped> FromStringList(List<PatchTitleString> list)
        {
            var result = new List<PatchTitleTyped>();
            foreach (var s in list) result.Add(s.ToPatchTitleTyped());
            return result;
        }

        public void WriteJson(Utf8JsonWriter w)
        {
            w.WriteStartObject();
            w.WriteString("title", this.title);
            w.WriteEndObject();
        }

        public string ToJSON()
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer)) { WriteJson(w); }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static PatchTitleTyped FromJsonElement(JsonElement m)
        {
            return new PatchTitleTyped(
                Json.AsString(Json.Require(m, "title"), "title")
            );
        }

        public static PatchTitleTyped FromJSON(string json)
        {
            return FromJsonElement(Json.Parse(json));
        }

        public static string ToJSONList(List<PatchTitleTyped> list)
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer))
            {
                w.WriteStartArray();
                foreach (var item in list) item.WriteJson(w);
                w.WriteEndArray();
            }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static List<PatchTitleTyped> FromJSONList(string json)
        {
            var result = new List<PatchTitleTyped>();
            var root = Json.Parse(json);
            Json.RequireArray(root, "PatchTitleTyped");
            foreach (var e in root.EnumerateArray()) result.Add(FromJsonElement(e));
            return result;
        }

        public override string ToString()
        {
            return $"title={title}";
        }

        public override bool Equals(object? obj)
        {
            if (obj is not PatchTitleTyped other) return false;
            return object.Equals(this.title, other.title);
        }

        public override int GetHashCode()
        {
            var h = new System.HashCode();
            h.Add(this.title);
            return h.ToHashCode();
        }
    }
}

File: Java/SpectableJavaExamplesTest/src/test/java/spectable/common/PatchTitleTyped.java
package spectable.common;

import java.util.ArrayList;
import java.util.List;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Objects;
import production.*;
import records.*;
import calculator.*;

public class PatchTitleTyped {
    public String title;

    public PatchTitleTyped(String title) {
        this.title = title;
    }

    public PatchTitleTyped(PatchTitleString s) {
        this.title = s.title;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof PatchTitleTyped)) return false;
        PatchTitleTyped that = (PatchTitleTyped) o;
        return Objects.equals(title, that.title);
    }

    @Override
    public int hashCode() {
        return Objects.hash(title);
    }

    public PatchTitleString toPatchTitleString() {
        return new PatchTitleString(String.valueOf(title));
    }

    public static List<PatchTitleTyped> fromStringList(List<PatchTitleString> list) {
        List<PatchTitleTyped> result = new ArrayList<>();
        for (PatchTitleString s : list) result.add(new PatchTitleTyped(s));
        return result;
    }

    public static List<PatchTitleString> toStringList(List<PatchTitleTyped> list) {
        List<PatchTitleString> result = new ArrayList<>();
        for (PatchTitleTyped t : list) result.add(t.toPatchTitleString());
        return result;
    }

    /** Internal plumbing for {@link Json}; use toJSON() for JSON text. */
    public Map<String, Object> toJsonValue() {
        Map<String, Object> m = new LinkedHashMap<String, Object>();
        m.put("title", title);
        return m;
    }

    public String toJSON() { return Json.write(toJsonValue()); }

    /** Internal plumbing for {@link Json}; use fromJSON(String) for JSON text. */
    public static PatchTitleTyped fromJsonValue(Map<String, Object> m) {
        if (m == null) return null;
        return new PatchTitleTyped(
            Json.asString(Json.require(m, "title"), "title"));
    }

    public static PatchTitleTyped fromJSON(String json) {
        return fromJsonValue(Json.parseObject(json));
    }

    public static String toJSONList(List<PatchTitleTyped> list) {
        List<Object> arr = new ArrayList<Object>();
        if (list != null)
            for (PatchTitleTyped item : list) arr.add(item == null ? null : item.toJsonValue());
        return Json.write(arr);
    }

    public static List<PatchTitleTyped> fromJSONList(String json) {
        List<PatchTitleTyped> result = new ArrayList<PatchTitleTyped>();
        for (Object e : Json.parseArray(json))
            result.add(fromJsonValue(Json.asObject(e, "PatchTitleTyped")));
        return result;
    }
}
